include/dxc/Support/dxcapi.use.h

```diff
@@ -85,7 +129,7 @@ class DxcDllSupport {
     other.m_createFn2 = nullptr;
   }
 
-  ~DxcDllSupport() { Cleanup(); }
+  ~SpecificDllLoader() { Cleanup(); }
 
   HRESULT Initialize() {
     return InitializeInternal(kDxCompilerLib, "DxcCreateInstance");
```

Comment:
Having this particular implementation of `Initialize()` in `SpecificDllLoader` is kind of irksome. This means that `SpecificDllLoader` really does want to load `dxcompiler.dll` in particular, even in cases where we're never using it for that dll. It would be nice for `DxcompilerLoader` to be a subclass akin to `HlslIntellisenseSupport` that specifically bakes in the defaults.

I don't think this is really possible without either removing `Initialize()` from the interface or explicitly implementing `SpecificDllLoader::Initialize()` to throw a runtime error though.

Reply:
What I'm saying is is "do we even want an Initialize()` function with no parameters on SpecificDllLoader? What if `SpecificDllLoader` only had `InitializeForDll()` and to use it you had to specify which dll you wanted to load?

For convenience, we could subclass this with `DxcompilerDllLoader` that specifically loads `dxcompiler.dll`, much like we do with `HlslIntellisenseSupport`. That object could reasonably have a `Initialize()` that knows what Dll to load. This would match the `Initialize()` on the external validator class that knows which two Dlls it's going to load.